src/streamlink/plugins/vkplay.py

```diff
@@ -0,0 +1,89 @@
+
+from streamlink.plugin import Plugin, pluginmatcher
+from streamlink.plugin.api import validate
+from streamlink.stream.hls import HLSStream
+
+log = logging.getLogger(__name__)
+
+
+@pluginmatcher(re.compile(
+    r"https?://vkplay\.live/(?P<channel_name>\w+)/?$",
+))
+class VKplay(Plugin):
+    API_URL = "https://api.vkplay.live/v1"
+
+    def _get_streams(self):
+        self.author = self.match.group("channel_name")
+        log.debug(f"Channel name: {self.author}")
+
+        data = self.session.http.get(
+            f"{self.API_URL}/blog/{self.author}/public_video_stream",
+            headers={"Referer": self.url},
+            acceptable_status=(200, 404),
+            schema=validate.Schema(
+                validate.parse_json(),
+                validate.any(
+                    validate.all(
+                        {"error": str, "error_description": str},
+                        validate.get("error_description"),
+                    ),
+                    validate.all(
+                        {
+                            "category": {
+                                "title": str,
+                            },
+                            "title": str,
+                            "data": validate.any(
+                                [
+                                    validate.all(
+                                        {
+                                            "vid": str,
+                                            "playerUrls": [
+                                                validate.all(
+                                                    {
+                                                        "type": str,
+                                                        "url": validate.any("", validate.url()),
+                                                    },
+                                                    validate.union_get("type", "url"),
+                                                ),
+                                            ],
+                                        },
+                                        validate.union_get("vid", "playerUrls"),
+                                    ),
+                                ],
+                                [],
+                            ),
+                        },
+                        validate.union_get(
+                            ("category", "title"),
+                            "title",
+                            ("data", 0),
+                        ),
+                    ),
+                ),
+            ),
+        )
+        if type(data) is str:
+            log.error(data)
+            return
+
+        self.category, self.title, streamdata = data
+        if not streamdata:
+            return
+
+        self.id, streams = streamdata
+
+        for streamtype, streamurl in streams:
+            if streamtype == "live_hls":
+                return HLSStream.parse_variant_playlist(self.session, streamurl)
```

Comment:
Not that it seems to cause much of an issue in practice, but `streamurl` could be an empty string here.  I guess in practice that might not actually happen when `streamtype == "live_hls"` anyway, so I don't know if you'd consider it worth checking for.

Reply:
I had a check for this before I commited my changes but then removed it. Let's add it back, though.